src/main/java/com/minecolonies/coremod/entity/pathfinding/AbstractWalkToProxy.java

```diff
@@ -0,0 +1,310 @@
+package com.minecolonies.coremod.entity.pathfinding;
+
+import com.minecolonies.api.util.BlockPosUtil;
+import com.minecolonies.api.util.EntityUtils;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.util.math.BlockPos;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Abstract class for the walkToProxy.
+ */
+public abstract class AbstractWalkToProxy
```

Comment:
Sure I moved it to the APi Package, the abstractWalkToProxy has no dependencies on minecolonies.
Only on:
import com.minecolonies.api.util.BlockPosUtil;
import com.minecolonies.api.util.EntityUtils;
import net.minecraft.entity.EntityLiving;
import net.minecraft.util.math.BlockPos;